MoreLinq.Test/IndexByTest.cs

```diff
@@ -0,0 +1,100 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 Leandro F. Vieira (leandromoh). All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class IndexByTest
+    {
+        [Test]
+        public void IndexBySimpleTest()
+        {
+            var source = new[] { "ana", "beatriz", "carla", "bob", "davi", "adriano", "angelo", "carlos" };
+            var result = source.IndexBy(x => x.First(), (item, key, index) => (item, key, index));
```

Comment:
`(item, key, index) => (item, key, index)` can be shortened to `ValueTuple.Create` but if the goal here is to name the elements, then please use PascalCase for consistency or just an anonymous-typed object like `(e, k, i) => new { Item = e, Key = k, Index = i }`.